source/aws_profile.c

```diff
@@ -601,28 +607,90 @@ static void s_aws_profile_collection_destroy_internal(struct aws_profile_collect
     aws_hash_table_clean_up(&profile_collection->profiles);
     aws_mem_release(profile_collection->allocator, profile_collection);
 }
+AWS_STATIC_STRING_FROM_LITERAL(s_profile_token, "profile");
+AWS_STATIC_STRING_FROM_LITERAL(s_sso_session_token, "sso-session");
 
 const struct aws_profile *aws_profile_collection_get_profile(
     const struct aws_profile_collection *profile_collection,
     const struct aws_string *profile_name) {
     struct aws_hash_element *element = NULL;
-    aws_hash_table_find(&profile_collection->profiles, profile_name, &element);
+    if (aws_hash_table_find(&profile_collection->profiles, s_profile_token, &element) || element == NULL) {
+        return NULL;
+    }
+    struct aws_section_collection *collection = element->value;
 
-    if (element == NULL) {
+    struct aws_hash_element *profile = NULL;
+    aws_hash_table_find(&collection->sections, profile_name, &profile);
```

Comment:
nit: hash table find seems inconsistent. first one checks for error, second one ignores it

Reply:
Thanks, fixed by refactor.